src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceConditionService.kt

```diff
@@ -255,6 +255,26 @@ class LicenceConditionService(
     }
   }
 
+  @Transactional
+  fun deleteAdditionalConditions(licenceId: Long, conditionIds: List<Long>) {
+    val licenceEntity = licenceRepository
+      .findById(licenceId)
+      .orElseThrow { EntityNotFoundException("$licenceId") }
```

Comment:
you already have the licence so don't need to re-load it here. 

Would suggest, moving the deleteAdditionalCondition function over to delegate to this function
```kt
fun deleteAdditionalCondition(licenceId: Long, conditionId: Long) 
  val licenceEntity = licenceRepository
      .findById(licenceId)
      .orElseThrow { EntityNotFoundException("$licenceId") }
  deleteAdditionalConditions(licenceEntity, listOf(conditionId))
}
```